externals/kyuubi-spark-sql-engine/src/test/scala/org/apache/kyuubi/engine/spark/operation/SparkOperationProgressSuite.scala

```diff
@@ -0,0 +1,130 @@
+      val req = new TExecuteStatementReq()
+      req.setStatement(sql)
+      req.setRunAsync(true)
+      req.setSessionHandle(handle)
+      val resp = client.ExecuteStatement(req)
+      var checkFlag1 = false
+      var checkFlag2 = false
+      var checkFlag3 = false
+      val initStageId = initJobId
+      eventually(Timeout(25.seconds)) {
+        val statusReq = new TGetOperationStatusReq(resp.getOperationHandle)
+        val statusResp = client.GetOperationStatus(statusReq)
+        val headers = statusResp.getProgressUpdateResponse.getHeaderNames
+        val progress = statusResp.getProgressUpdateResponse.getProgressedPercentage
+        val rows = statusResp.getProgressUpdateResponse.getRows
+        val footerSummary = statusResp.getProgressUpdateResponse.getFooterSummary
+        val status = statusResp.getProgressUpdateResponse.getStatus
+        assertResult(Seq(
+          "STAGES",
+          "ATTEMPT",
+          "STATUS",
+          "TOTAL",
+          "COMPLETED",
+          "RUNNING",
+          "PENDING",
+          "FAILED"))(headers.asScala)
+        assert(rows.size() == 1)
+        try {
+          progress match {
+            case 0.0 =>
+              assert(Seq(s"Stage-$initStageId ", "0", "PENDING", "2", "0", "0", "2", "0")
+                == rows.get(0).asScala ||
+                Seq(s"Stage-$initStageId ", "0", "RUNNING", "2", "0", "1", "1", "0")
+                == rows.get(0).asScala ||
+                Seq(s"Stage-$initStageId ", "0", "RUNNING", "2", "0", "2", "0", "0")
+                == rows.get(0).asScala)
+              assert("STAGES: 00/01" === footerSummary)
+              assert(TJobExecutionStatus.IN_PROGRESS === status)
+              checkFlag1 = true
+            case 0.5 =>
+              assertResult(Seq(
+                s"Stage-$initStageId ....",
+                "0",
+                "RUNNING",
+                "2",
+                "1",
+                "1",
+                "0",
+                "0"))(
+                rows.get(0).asScala)
+              assert("STAGES: 00/01" === footerSummary)
+              assert(TJobExecutionStatus.IN_PROGRESS === status)
+              checkFlag2 = true
+            case 1.0 =>
+              assertResult(Seq(
+                s"Stage-$initStageId ........",
+                "0",
+                "FINISHED",
+                "2",
+                "2",
+                "0",
+                "0",
+                "0"))(
+                rows.get(0).asScala)
+              assert("STAGES: 01/01" === footerSummary)
+              checkFlag3 = true
+          }
+        } catch {
+          case tfe: TestFailedException =>
+            // scalastyle:off println
+            println(s"progress: $progress")
+            println(s"rows: $rows")
+            println(s"footerSummary: $footerSummary")
+            println(s"status: $status")
+            // scalastyle:on println
```

Comment:
remove this, if needed we can use `info`

Reply:
> remove this, if needed we can use `info`

It is only used to debug this test case, I will remove it.